be/src/util/compression/block_compression.cpp

```diff
@@ -252,55 +279,63 @@ class Lz4HadoopBlockCompression : public Lz4BlockCompression {
 
 class Lz4fBlockCompression : public BlockCompressionCodec {
 public:
+    Lz4fBlockCompression() : BlockCompressionCodec(CompressionTypePB::LZ4_FRAME) {}
+
     static const Lz4fBlockCompression* instance() {
         static Lz4fBlockCompression s_instance;
         return &s_instance;
     }
 
     ~Lz4fBlockCompression() override = default;
 
-    Status compress(const Slice& input, Slice* output) const override {
-        auto compressed_len = LZ4F_compressFrame(output->data, output->size, input.data, input.size, &_s_preferences);
-        if (LZ4F_isError(compressed_len)) {
-            return Status::InvalidArgument(strings::Substitute("Fail to do LZ4FRAME compress frame, msg=$0",
-                                                               LZ4F_getErrorName(compressed_len)));
-        }
-        output->size = compressed_len;
-        return Status::OK();
+    Status compress(const Slice& input, Slice* output, bool use_compression_buffer, size_t uncompressed_size,
```

Comment:
1. The following three files also use compression. You have to change it also.
```
exec/decompressor.cpp:44
exec/pipeline/exchange/exchange_sink_operator.cpp:327
runtime/data_stream_sender.cpp:459
```

2. If you change the previous files, the query latency will be improved also when high QPS due to the reduce of conflict in tcmalloc. It has to test it in the QA performance test.


3. `exec/vectorized/file_scanner.cpp` this file use another decompression in `exec/decompressor.h`. You can unify it in another pr.